merkletree/proof.go

```diff
@@ -80,38 +80,51 @@ func (ap *AuthenticationPath) authPathHash() []byte {
 	return hash
 }
 
-func (ap *AuthenticationPath) verifyBinding(key, value []byte) bool {
+// VerifyBinding verifies both the value and the commitment
+// of the proof node of the authentication path.
+//
+// This should be called only when ap is a proof of inclusion.
+func (ap *AuthenticationPath) VerifyBinding(key, value []byte) bool {
```

Comment:
Yes, rather than the comment,

```
// This should be called only when ap is a proof of inclusion.
```

why not just assert `ap.ProofType() == ProofOfInclusion` as well?